aiohttp/helpers.py

```diff
@@ -654,3 +654,35 @@ def content_length(self, *, _CONTENT_LENGTH=hdrs.CONTENT_LENGTH):
 
         if content_length:
             return int(content_length)
+
+
+def format_parameter_value(value):
+    """Wrap a parameter value in quotes, if necessary.
+
+    Expects and returns a byte-string.
+    """
+    # Refer to RFCs 7231, 7230, 5234.
+    #
+    # parameter      = token "=" ( token / quoted-string )
+    # token          = 1*tchar
+    # quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+    # qdtext         = HTAB / SP / %x21 / %x23-5B / %x5D-7E / obs-text
+    # obs-text       = %x80-FF
+    # quoted-pair    = "\" ( HTAB / SP / VCHAR / obs-text )
+    # tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+    #                  / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+    #                  / DIGIT / ALPHA
+    #                  ; any VCHAR, except delimiters
+    # VCHAR           = %x21-7E
+    valid_tchar_regex = br"^[!#$%&'*+\-.^_`|~\w]+$"
+    if re.match(valid_tchar_regex, value):
+        return value
+
+    invalid_qdtext_char_regex = br"[\x00-\x08\x0A-\x1F\x7F]"
+    if re.match(invalid_qdtext_char_regex, value):
+        raise ValueError("parameter value contains invalid characters")
+
+    quoted_value_content = value.replace(b'\\', b'\\\\')  # %x5C
+    quoted_value_content = value.replace(b'"', b'\\"')  # %x22
+
+    return b'"' + quoted_value_content + b'"'
```

Comment:
Why bytes here in and out?

Reply:
The HTTP spec operates on bytes, which is why I designed this helper function to follow that. The only reason why we need to convert back to a string is because the underlying Payload API requires a string for header values. Or at least I think so. If it doesn't, I'll be glad to remove the conversion back to a unicode string. 

 I could of course special-case the function for just our situation here, but it should go into multipart.py then.

Reply:
Also, we need to use bytes in the actual multipart body, so we should try to convert exactly that onto a potentially quoted parameter value.